scripts/examples/esp_example.sh

```diff
@@ -36,14 +36,17 @@ source "$IDF_PATH/export.sh"
 for sdkconfig in "$root"/sdkconfig*.defaults; do
     # remove root path to get sdkconfig*.defaults name
     sdkconfig_name=${sdkconfig#"$root"/}
+    if [ "$sdkconfig_name" == "sdkconfig_c3devkit.defaults" ]; then
+        idf_target="esp32c3"
+    else
+        idf_target="esp32"
+    fi
+
     rm -f "$root"/sdkconfig
     (
         cd "$root"
+        idf.py set-target "$idf_target"
         idf.py -D SDKCONFIG_DEFAULTS="$sdkconfig_name" build
-    )
-    (
-        cd "$root"
-        idf.py build "$@"
```

Comment:
This is removed as a part of the PR #7910 .

Reply:
Rebased